octopoes/octopoes/models/ooi/reports.py

```diff
@@ -39,10 +39,25 @@ class Report(OOI):
 
     observed_at: datetime
     parent_report: Reference | None = ReferenceField("Report", default=None)
+    report_recipe: Reference | None = ReferenceField("ReportRecipe", default=None)
     has_parent: bool
 
     _natural_key_attrs = ["report_id"]
 
     @classmethod
     def format_reference_human_readable(cls, reference: Reference) -> str:
         return f"Report {reference.tokenized.report_id}"
+
+
+class ReportRecipe(OOI):
+    recipe_id: UUID
+    # object can be the total objects or 1 object, but then the ooi itself.
+    report_name_format: str
+    subreport_name_format: str
```

Comment:
Should maybe be optional? Not sure.

Reply:
I think we must set a name for the report and subreports, otherwise the PDF headings would be incomplete and not visible. Also not clear in json export which report belongs to what.

Reply:
Okay